core/contract/src/test/java/org/eclipse/dataspaceconnector/contract/negotiation/ContractNegotiationIntegrationTest.java

```diff
@@ -60,40 +69,47 @@ void testNegotiation_initialOfferAccepted() throws Exception {
                 .build();
         consumerManager.initiate(request);
 
-        // Wait for negotiation to finish with time out at 15 seconds
-        var success = countDownLatch.await(15, TimeUnit.SECONDS);
 
-        assertThat(success).isTrue();
+        await().atMost(DEFAULT_TEST_TIMEOUT)
+                .pollInterval(DEFAULT_POLL_INTERVAL)
+                .untilAsserted(() -> {
+                    var consumerNegotiation = consumerStore.find(consumerNegotiationId);
+                    var providerNegotiation = providerStore.findForCorrelationId(consumerNegotiationId);
+
+                    // Assert that provider and consumer have the same offers and agreement stored
+                    assertThat(consumerNegotiation).isNotNull();
+                    assertThat(providerNegotiation).isNotNull();
+                    assertThat(consumerNegotiation.getState()).isEqualTo(CONFIRMED.code());
+                    assertThat(consumerNegotiation.getContractOffers()).hasSize(1);
+                    assertThat(consumerNegotiation.getContractOffers().size()).isEqualTo(providerNegotiation.getContractOffers().size());
```

Comment:
seems strange given previous line. Is it meant to revert the assertion?

Reply:
I don't really know, but I would assume that both things are to be verified, the size being equal to 1, and the sizes of both contract offer lists being equal

Reply:
ok to resolve